### DISCUSSION TITLE
French Tacotron2 DDC TTS model with HifiGan2 very noisy

### DISCUSSION THREAD

**Discussion by lpierron** [coqui-ai/TTS#458] [General Q&A]
I trained a French TTS model with Tacotron2 DDC from MAI-Labs. I'm using Coqui-TTS v.0.0.12.

I tried TTS with the vocoder `vocoder_models--en--ljspeech--hifigan_v2`, as dowloaded fro Coqui-TTS.

The resulting audio file is very noisy as you can hear: [https://sndup.net/2t9d](https://sndup.net/2t9d)

You can find as a gist my `config.json` and `vocoder_config.json`: [https://gist.github.com/lpierron/6c56302eb628ee6a86363daa08e5fa63](https://gist.github.com/lpierron/6c56302eb628ee6a86363daa08e5fa63)

Any idea to solve the noisy problem ?

I tried using another Vocoder (Melgan one) and there is no noise, but the voice is hoarse as you can her: [https://sndup.net/4jgn](https://sndup.net/4jgn)

**Comment by a-froghyar** [comment]
Are the griffin-lim samples (straight audio from Tacotron without the vocoder) also noisey?

**Comment by lpierron** [comment]
No the griffin-lim samples are OK: https://sndup.net/3t2b

**Comment by lpierron** [comment]
There is something strange with the **_Melgan_** and _**HifiGan**_ vocoders, their outputs are in **25 bits**, why?

**Comment by erogol** [comment]
You can use the preferred vocoder by just calling 

```console 
$ tts --model_name tts_models/fr/mai/tacotron2-DDC --text "This is text." 
```

Not all vocoders are compatible with all the models.

**Comment by lpierron** [comment]
I don't agree, the preferred french **text2feat** model is **_bad_**, so I trained a new text2feat model.
I want to understand, why my  french `text2feat ` model trained with the same parameters as your ljspeech english model cannot be used with the HifiGan2 vocoder.
You didn't answer why the vocoders outputs are in 25 bits.

**Comment by erogol** [comment]
check the audio parameters in the vocoder and your model and see if they match. If they don't you can't use that vocoder with your model. 

It is not a bug this thread needs to be in Discussions.

**Reply by lpierron** [reply]
OK, sorry for the trouble in Issues.

The differences in audio parameters are:

|                 | HiFiGan2 | French DDC          |
|-----------------|----------|---------------------|
| log_func        | np.log   |                     |
| do_trim_silence | false    | true                |
| signal_norm     | false    | true                |
| stats_path      | null     | "./scale_stats.npy" |

Maybe the `log_func` makes the difference ?

**Reply by erogol** [reply]
yes. The default is log10 and the vocoder uses log. You can try fine-tuning the model with your own audio params.

**Reply by valvin1** [reply]
Hello, I'm just discovering Coqui and I'm testing french model and having noisy quality by default. It looks that by default vocoder used is not compatible with tts : https://github.com/coqui-ai/TTS/blob/dev/TTS/.models.json#L160
I tried to test different vocoder and the only which give an almost correct result is `vocoder_models/universal/libri-tts/wavegrad` but it is very resource expensive.
Am I doing something wrong? The audio quality between english and french is very important and it could be interesting if there was something almost similar or approaching :)
